### PR TITLE
added spec.md to ONS-UK-trade-in-services-by-business-characteristics

### PR DESCRIPTION
**Pull request by rg5051**
Hi, could you read the spec.md please? It's a first draft so it won't be complete yet but I want to see if I'm on the right lines.

Apologies for the dodgy formatting. I copy-pasted one of Leigh's into a text editor to work on it and it hasn't rendered some of the characters properly.

My main questions and points are:

- do you agree with my assessment about how the tabs can be merged?
- there is nothing in the Stage 1 Transform section because I think that is supposed to be filled in by a DE once they've done that
- I'm not sure what to do with the data marking column (if anything) and what order (if any) the columns go in
- I'm not sure how I'm supposed to update the info.json code lists.

Finally, as this is not a finished spec, I'm not expecting it to be merged into master just yet.

Might be best to call to discuss?

### CODE REVIEW

File: datasets/ONS-UK-trade-in-services-by-business-characteristics/spec.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+{\rtf1\ansi\ansicpg1252\cocoartf1671\cocoasubrtf600
```
Comment (robons) [inline datasets/ONS-UK-trade-in-services-by-business-characteristics/spec.md:1]: I'm not sure what's going on here, but all this stuff with backslashes isn't markdown and makes the document really hard to read.

File: datasets/ONS-UK-trade-in-services-by-business-characteristics/spec.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+		Columns go here\
+\
+#### DE Stage one notes \
+Notes go here.\
+\
+### Stage 2 - Harmonisation\
+\
+Exports (\'a3m) and Imports (\'a3m) should be combined into a dimension and the values should go in the \'91Value\'92 column.\
+\
+I think the six data tabs can be combined into a single cube: 2016, 2016 Industry Totals, 2017, 2017 Industry Totals, 2018, 2018 Industry Totals.\
```
Comment (robons) [inline datasets/ONS-UK-trade-in-services-by-business-characteristics/spec.md:38]: If all of the tabs are going to be combined together you need to ensure they all have the same columns first. You might do this by adding `Ownership` to sheets which don't have any ownership information and specifying a new 'All' ownership type in the codelist you'll create.

You'll need to do the same with `region` and `business size`.

File: datasets/ONS-UK-trade-in-services-by-business-characteristics/spec.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+#### Table Structure\
+\
+		Columns go here\
+\
+#### DE Stage one notes \
+Notes go here.\
+\
+### Stage 2 - Harmonisation\
+\
+Exports (\'a3m) and Imports (\'a3m) should be combined into a dimension and the values should go in the \'91Value\'92 column.\
```
Comment (robons) [inline datasets/ONS-UK-trade-in-services-by-business-characteristics/spec.md:36]: In order to represent this you'll need to add the `flow` dimension - the codelist is already defined for you in https://github.com/GSS-Cogs/ref_common/blob/pmd4/codelists/flow.csv

File: datasets/ONS-Quarterly-country-and-regional-GDP/info.json [hunk 4]
```diff
@@ -15,7 +15,6272 @@
         "stage": [
             "To Do"
         ],
-        "main_issue": 85
+        "main_issue": 85,
```
Comment (robons) [inline datasets/ONS-Quarterly-country-and-regional-GDP/info.json:18]: This info.json is in the wrong directory. Can you revert your changes here?

File: datasets/ONS-UK-trade-in-services-by-business-characteristics/info.json [hunk 5]
```diff
@@ -18,7 +18,1134 @@
         "stage": [
             "To Do"
         ],
-        "main_issue": 89
+        "main_issue": 89,
+        "transformStage": [
```
Comment (robons) [inline datasets/ONS-UK-trade-in-services-by-business-characteristics/info.json:22]: Can you remove the `transformStage` stuff? It makes it very hard to check if you've added any `columns` to this file. 

Have you started considering how your columns will link up to other external dimensions? Any thoughts on which measure and units you're going to use?

File: datasets/ONS-UK-trade-in-services-by-business-characteristics/spec.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+\
+I think the six data tabs can be combined into a single cube: 2016, 2016 Industry Totals, 2017, 2017 Industry Totals, 2018, 2018 Industry Totals.\
+\
+A possible dataset structure would be:\
```
Comment (robons) [inline datasets/ONS-UK-trade-in-services-by-business-characteristics/spec.md:40]: Have you considered what you're going to do with the `..` suppressed values?

**Review by robons — COMMENTED** [review]
I've left some comments suggesting where you may want to go from here. Feel free to ask questions. If you're unsure about anything, stick another comment on the PR or talk to me.

File: datasets/ONS-UK-trade-in-services-by-business-characteristics/spec.md (hunk 1 above)
Comment (rg5051) [reply datasets/ONS-UK-trade-in-services-by-business-characteristics/spec.md]: Done this.

File: datasets/ONS-UK-trade-in-services-by-business-characteristics/spec.md (hunk 3 above)
Comment (rg5051) [reply datasets/ONS-UK-trade-in-services-by-business-characteristics/spec.md]: Thanks - originally this was a 'Direction' dimension but I've changed that to 'Flow' in the spec.

**Pull request closed without merging**